pkg/commands/plan/plan_init.go

```diff
@@ -0,0 +1,243 @@
+	cli.BaseCommand
+
+	workingDir  string
+	entrypoints []string
+
+	flags.GitHubFlags
+	flags.RetryFlags
+
+	flagPullRequestNumber      int
+	flagDestRef                string
+	flagSourceRef              string
+	flagDirectories            []string
+	flagDeleteOutdatedComments bool
+	flagJSON                   bool
+
+	gitClient    git.Git
+	githubClient github.GitHub
+
+	// testFlagSetOpts is only used for testing.
+	testFlagSetOpts []cli.Option
+}
+
+func (c *PlanInitCommand) Desc() string {
+	return `Run the Terraform plan for a directory.`
+}
+
+func (c *PlanInitCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Initialize Guardian for running the Terraform plan process.
+`
+}
+
+func (c *PlanInitCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet(c.testFlagSetOpts...)
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("Command options")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "dest-ref",
+		Target:  &c.flagDestRef,
+		Example: "ref-name",
+		Usage:   "The destination GitHub ref name for finding file changes.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-ref",
+		Target:  &c.flagSourceRef,
+		Example: "ref-name",
+		Usage:   "The source GitHub ref name for finding file changes.",
+	})
+
+	f.StringSliceVar(&cli.StringSliceVar{
+		Name:    "directories",
+		Target:  &c.flagDirectories,
+		Example: "dir1,dir2,dir3",
+		Usage:   "The subset of directories to target for Terraform planning.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "delete-outdated-comments",
+		Target:  &c.flagDeleteOutdatedComments,
+		Default: true,
+		Usage:   "Delete outdated comments when Guardian plan is run multiple times for the same pull request.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "json",
+		Target:  &c.flagJSON,
+		Default: false,
+		Usage:   "Print the output directories in JSON format.",
+	})
+
+	return set
+}
+
+func (c *PlanInitCommand) Run(ctx context.Context, args []string) error {
+	logger := logging.FromContext(ctx)
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	c.workingDir = cwd
+
+	c.gitClient = git.NewGitClient(cwd)
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		github.WithMaxRetries(c.RetryFlags.FlagMaxRetries),
+		github.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+
+	if len(c.flagDirectories) == 0 {
+		c.flagDirectories = append(c.flagDirectories, cwd)
+	}
```

Comment:
This behavior seems a little weird? If you don't specify any directories, it defaults to the cwd? That should at least be documented.

Reply:
will adjust to using args